### PR TITLE
ICU-21176 Add aliases for terms "whitelist" and "blacklist" in data filter

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21176
- [x] Updated PR title and link in previous line to include Issue number
- [ ] Issue accepted
- [x] Tests included
- [x] Documentation is changed or added

### CODE REVIEW

File: .ci-builds/data-filter.json [hunk 1]
```diff
@@ -9,17 +9,22 @@
   },
 // Test mixed feature filter and resource filter
 // Exlude translit data so we can run test for ICU-20673
```
Comment (markusicu) [inline .ci-builds/data-filter.json:11]: pre-existing typo "Exlude"

File: .ci-builds/data-filter.json [hunk 2]
```diff
@@ -9,17 +9,22 @@
...
+// Also test for "whitelist" versus "include"
   "featureFilters": {
     "misc": {
       "whitelist": ["supplementalData"]
     },
-    "translit": "exclude"
+    "translit": "exclude",
+    "curr_tree": {
+      "filterType": "locale",
+      "include": ["my"]
```
Comment (markusicu) [inline .ci-builds/data-filter.json:20]: "include" should work but may not be the best option; see the doc for discussion

File: icu4c/source/python/icutools/databuilder/filtration.py [hunk 3]
```diff
@@ -83,10 +83,17 @@ def __init__(self, json_data):
...
+            self.is_whitelist = True
+            self.whitelist = json_data["include"]
+        elif "blacklist" in json_data:
             self.is_whitelist = False
             self.blacklist = json_data["blacklist"]
+        elif "exclude" in json_data:
+            self.is_whitelist = False
+            self.blacklist = json_data["exclude"]
+        else:
+            raise AssertionError("Need either whitelist or blacklist: %s" % str(json_data))
```
Comment (markusicu) [inline icu4c/source/python/icutools/databuilder/filtration.py:96]: needs update

File: icu4c/source/python/icutools/databuilder/filtration.py [hunk 4]
```diff
@@ -159,7 +166,12 @@ def match(self, file):
 
 class LocaleFilter(Filter):
     def __init__(self, json_data, io):
-        self.locales_requested = list(json_data["whitelist"])
+        if "whitelist" in json_data:
+            self.locales_requested = list(json_data["whitelist"])
+        elif "include" in json_data:
+            self.locales_requested = list(json_data["include"])
+        else:
+            raise AssertionError("You must include a whitelist in a locale filter")
```
Comment (markusicu) [inline icu4c/source/python/icutools/databuilder/filtration.py:174]: needs update

File: icu4c/source/python/icutools/databuilder/filtration_schema.json [hunk 5]
```diff
@@ -107,6 +107,26 @@
                     "required": ["blacklist"],
                     "additionalProperties": false
                 },
+                {
+                    "properties": {
+                        "filterType": {
+                            "$ref": "#/definitions/blacklistWhitelistFilterTypes"
```
Comment (markusicu) [inline icu4c/source/python/icutools/databuilder/filtration_schema.json:113]: can we rename this ...FilterTypes?

File: .ci-builds/data-filter.json (hunk 1 above)
Comment (sffc) [reply .ci-builds/data-filter.json:11]: Done

File: .ci-builds/data-filter.json (hunk 2 above)
Comment (sffc) [reply .ci-builds/data-filter.json:20]: Changed to includelist per discussion

File: icu4c/source/python/icutools/databuilder/filtration.py (hunk 3 above)
Comment (sffc) [reply icu4c/source/python/icutools/databuilder/filtration.py:96]: Done

File: icu4c/source/python/icutools/databuilder/filtration.py (hunk 4 above)
Comment (sffc) [reply icu4c/source/python/icutools/databuilder/filtration.py:174]: Done

File: icu4c/source/python/icutools/databuilder/filtration_schema.json (hunk 5 above)
Comment (sffc) [reply icu4c/source/python/icutools/databuilder/filtration_schema.json:113]: Done

File: .ci-builds/data-filter.json [hunk 6]
```diff
@@ -8,18 +8,29 @@
     ]
   },
 // Test mixed feature filter and resource filter
-// Exlude translit data so we can run test for ICU-20673
+// Exclude translit data so we can run test for ICU-20673
+// Also test for "whitelist" versus "include"
```
Comment (markusicu) [inline .ci-builds/data-filter.json:12]: "includelist"

**Pull request merged**